syntax = "proto3";

package products;

service products {
    rpc get_product(GetProduct) returns (Product);
    rpc create_product(Product) returns (Product);
    rpc list_products(GetProducts) returns (Products);
    rpc update_products_inventory(UpdateInventoryRequest) returns (UpdateInventoryResponse);
}

message Product {
    string id = 1;
    string title = 2;
    int32 passenger_capacity = 3;
    int32 maximum_speed = 4;
    int32 in_stock = 5;
}

message Products {
    repeated Product products = 1;
}

message GetProduct {
    string id = 1;
}

message GetProducts {}

message UpdateInventoryRequest {
    repeated UpdateProductInventoryRequestDetails updateproductinventorydetails = 1;
}

message UpdateProductInventoryRequestDetails{
    string id = 1;
    int32 quantity = 2;
}

message UpdateInventoryResponse {
    repeated UpdateInventoryResponseDetails updateproductinventoryresponse = 1;
}

message UpdateInventoryResponseDetails {
    string id = 1;
    bool isupdated = 2;
}